src/app/transactions/StakingTransactionForm.tsx

```diff
@@ -88,14 +96,22 @@ export function StakingTransactionForm({
         format: "json", // FIXME Not always the default, should come from chains config
       };
 
+      // Handle auto-setting of sender for unstake or claim rewards based on selected staking position
+      if (
+        (mode === TransactionMode.UNDELEGATE ||
+          mode === TransactionMode.CLAIM_REWARDS) &&
+        selectedStakingPosition
+      ) {
+        transactionData.sender = selectedStakingPosition.addresses[0]; // Automatically use the first address from staking position
+      }
+
       if (formInput.amount && !formInput.useMaxAmount) {
         transactionData.amount = amountToSmallestUnit(
           formInput.amount.toString(),
           decimals
         );
       }
 
-      // FIXME Hack to be able to provide the pubKey, probably better to refacto
```

Comment:
@fabrice-adamik Please please pay attention to comments :pray: ChatGPT seems to not respect them a lot :stuck_out_tongue: (and a lot of the ones it adds actually bring little value, btw)

I'm going to put this one back. What you can do next times, is review your own PR here on GitHub. It's a good practice that I often do myself :slightly_smiling_face: will be easier to spot comment changes that way.